QgisModelBaker/gui/basket_manager.py

```diff
@@ -17,51 +17,196 @@
+            edit_basket_dialog = EditBasketDialog(
+                self, self.db_connector, selected_basket_settings
+            )
+            edit_basket_dialog.exec_()
+
+            # Refresh existing baskets in basket manager after creation
+            self.baskets_panel.bid_model.load_basket_config(
+                self.db_connector, self.datasetname
+            )
+
+    def _delete_basket(self) -> None:
+        if self._valid_selection():
+            if not self.db_connector.get_basket_handling():
+                QMessageBox.warning(
+                    self,
+                    self.tr("Delete Basket"),
+                    self.tr(
+                        "Delete baskets is only available for database schemas created with --createBasketCol parameter."
+                    ),
+                    QMessageBox.Close,
+                )
+                return
+
+            if (
+                QMessageBox.warning(
+                    self,
+                    self.tr("Delete Basket"),
+                    self.tr(
+                        "Deleting a Basket will also delete all the data it contains. This operation cannot be reverted.\n\nAre you sure you want to proceed?"
+                    ),
+                    QMessageBox.No | QMessageBox.Yes,
+                )
+                == QMessageBox.Yes
+            ):
+                basket_config = self.baskets_panel.selected_basket_settings()
+                res, msg = self._do_delete_basket(basket_config)
+
+                if res:
+                    # After deletion, make sure canvas is refreshed
+                    self._refresh_map_layers()
+
+                    # Refresh existing baskets in basket manager after deletion
+                    self.baskets_panel.bid_model.load_basket_config(
+                        self.db_connector, self.datasetname
+                    )
+
+                warning_box = QMessageBox(self)
+                warning_box.setIcon(
+                    QMessageBox.Information if res else QMessageBox.Warning
+                )
+                warning_box.setWindowTitle(self.tr("Delete Basket"))
+                warning_box.setText(msg)
+                warning_box.exec_()
+
+    def _refresh_map_layers(self):
+        # Refresh layer data sources and also their symbology (including feature count)
+        layer_tree_view = self.iface.layerTreeView()
+        for tree_layer in QgsProject.instance().layerTreeRoot().findLayers():
+            layer = tree_layer.layer()
+            layer.dataProvider().reloadData()
+            layer_tree_view.refreshLayerSymbology(layer.id())
+
+    def _do_delete_basket(self, basket_config):
+        # Keep original values just in case we need to go back to them
+        original_dataset_id = basket_config["dataset_t_id"]
+        original_dataset_name = basket_config["datasetname"]
+
+        # Create temporary dataset
+        tmp_dataset_id, tmp_dataset_name = "", "_tmp_dataset_tmp_"
```

Comment:
Good idea for a workaround. Still I'll ask Claude...

Reply:
Yeap, implemented that way to have something functional within budget. Let's see what Claude tells us.
If it's supported by ili2db in the future, it'll be straightforward to switch to it.